### PR TITLE
Feature/revamp card

### PR DESCRIPTION
**Pull request by marukosy124**
**Updates**
- To differentiate profile card and post card, referrer and referee card will share similar UI, while referrer post and referee post card will share another similar UI
- To maintain the UI consistency throughout the application, all cards share some common UI components such as the UI display for company name, industry, location and years of experience
- Fixed bug #128 (as this issue caused UI bugs related to the card component as well, so I fixed here together with the revamp)
- For referrer and referee card, click the avatar/ username/ 查看用戶檔案 will go to the profile page
- For referrer and referee post card:
   - click the avatar will go to the creator's profile page
   - click the card will not have any effect now as there is no post details page yet (will add that later)
   - show all the details of a post
   - if the description is too long, click the collapse button can see the full text
   
**✨✨Any opinion on the UI design is welcomed✨✨**

### CODE REVIEW

File: client/components/customized-ui/info-display/industry.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import { Icons } from "@/components/icons"
+
+interface IIndustryDisplayProps {
+  industry: string | null
+}
+const IndustryDisplay: React.FunctionComponent<
+  PropsWithChildren<IIndustryDisplayProps>
+> = ({ industry }) => {
+  return (
+    <div className="flex justify-start">
```
Comment (511234) [inline client/components/customized-ui/info-display/industry.tsx:12]: I think adding items-center will look nicer

File: client/components/customized-ui/info-display/location.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+interface ILocationDisplayProps {
+  city: string | null
+  province: string | null
+  country: string | null
+}
+const LocationDisplay: React.FunctionComponent<
+  PropsWithChildren<ILocationDisplayProps>
+> = ({ city, province, country }) => {
+  return (
+    <div className="flex justify-start">
```
Comment (511234) [inline client/components/customized-ui/info-display/location.tsx:15]: adding items-center will look nicer

File: client/components/customized-ui/info-display/years-of-experience.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { Icons } from "@/components/icons"
+
+interface IYearsOfExperienceDisplayProps {
+  yearOfExperience: number | null
+}
+const YearsOfExperienceDisplay: React.FunctionComponent<
+  PropsWithChildren<IYearsOfExperienceDisplayProps>
+> = ({ yearOfExperience }) => {
+  return (
+    <div className="flex justify-start">
```
Comment (511234) [inline client/components/customized-ui/info-display/years-of-experience.tsx:12]: adding items-center will look nicer

Comment (marukosy124) [reply client/components/customized-ui/info-display/years-of-experience.tsx]: Agree. Added to all of them, thanks

**Comment by Pjaijai** [comment]
Impressive work! I can hardly believe you've managed to code all the components. I sincerely appreciate your efforts. I'll merge this after we roll out version 1.3.0. By the way, I've left a few comments for you to review. Please check one check.

新UI很正式

**Comment by 511234** [comment]
Edit button is always shown on profile page. 
20231015_profile_edit

Added checking at https://github.com/Pjaijai/Referalah/pull/135/commits/32c1555b6b8c58e0c5e0fcc1b701d3736b626525

**Review by Pjaijai — APPROVED** [review]
LGTM

**Review by BenjamincyLeung — APPROVED** [review]

**Pull request merged**